numpy/core/src/_simd/_simd_inc_arg.h

```diff
@@ -0,0 +1,68 @@
+#ifndef NPY_SIMD
+    #error "Not a standalone header"
+#endif
+
+/************************************
+ ** Protected Definitions
+ ************************************/
+static int
+simd_arg_parse(PyObject *py_args, simd_arg *args, int args_n)
+{
+    Py_ssize_t cur_arg = 0;
+    if (py_args == NULL) {
+        if (args_n == 0) {
+            return 1;
+        }
+        goto wrong_size;
+    }
+    int arg_pos = 0;
+    if (!PyTuple_Check(py_args)) {
+        if (args_n != 0) {
+            goto wrong_size;
+        }
+        simd_arg *a = &args[0]; a->obj = py_args;
+        a->data = simd_obj2data(py_args, a->type);
+        if (PyErr_Occurred()) {
+            goto wrong_data;
+        }
+        return 1;
+    }
+    cur_arg = PyTuple_GET_SIZE(py_args);
+    if (cur_arg != args_n) {
+        goto wrong_size;
+    }
+    for (; arg_pos < args_n; ++arg_pos) {
+        simd_arg *a = &args[arg_pos];
+        a->obj = PyTuple_GET_ITEM(py_args, arg_pos);
+        if (a->obj == NULL) {
+            if (!PyErr_Occurred()) {
+                PyErr_Format(PyExc_RuntimeError,
+                    "unable to get argument '%d'", arg_pos+1
+                );
+            }
+            goto wrong_data;
+        }
```

Comment:
This can never happen `PyTuple_GET_ITEM` always succeeds and tuple objects never contain null.

Reply:
I wouldn't dare to avoid checking a NULL, another "just in case" I guess.
Are you sure it would be safe to remove it?

Reply:
Yes, I'm certain. If this type of thing worries you, you can always use `assert(a->obj)`.

Attempting to gracefully handle a function with a bug in it (ie, the caller) incurs an undesirable complexity cost, and makes it harder to actually fix that bug if it does exist.